napari/_qt/utils.py

```diff
@@ -235,16 +236,29 @@ def combine_widgets(
         return widgets.native  # type: ignore
     elif isinstance(widgets, QWidget):
         return widgets
-    elif is_sequence(widgets) and all(isinstance(i, QWidget) for i in widgets):
-        container = QWidget()
-        container.setLayout(QVBoxLayout() if vertical else QHBoxLayout())
-        for widget in widgets:
-            container.layout().addWidget(widget)
-        return container
-    else:
-        raise TypeError(
-            trans._('"widget" must be a QWidget or a sequence of QWidgets')
-        )
+    elif is_sequence(widgets):
+        # the same as above, compatibility with magicgui v0.2.0
+        widgets = [
+            i.native if isinstance(getattr(i, 'native', None), QWidget) else i
+            for i in widgets
+        ]
+        if all(isinstance(i, QWidget) for i in widgets):
+            container = QWidget()
+            container.layout = QVBoxLayout() if vertical else QHBoxLayout()
```

Comment:
> I'm not sure about the error. Should the layout be callable?

yeah: try to never do this :)
you're shadowing the [`QWidget.layout`](https://doc.qt.io/qt-5/qwidget.html#layout) attribute here... turning it from a function that returns the current layout, into an instance variable

if you want to store the layout on the instance, store it anywhere else  (`._layout` would work). but also note that you don't need to store it:

```
widget.setLayout(QVBoxLayout())
widget.layout()  # there's your layout
```